django/db/models/indexes.py

```diff
@@ -17,11 +25,15 @@ def __init__(self, *, fields=[], name=None, db_tablespace=None):
         if not fields:
             raise ValueError('At least one field is required to define an index.')
         self.fields = fields
-        # A list of 2-tuple with the field name and ordering ('' or 'DESC').
-        self.fields_orders = [
-            (field_name[1:], 'DESC') if field_name.startswith('-') else (field_name, '')
-            for field_name in self.fields
-        ]
+        self.fields_names = [field_name.lstrip('-') for field_name in self.fields if isinstance(field_name, str)]
+        self.expressions = []
+        for field in self.fields:
+            if isinstance(field, str):
+                expression = F(field[1:]).desc() if field.startswith('-') else F(field)
```

Comment:
Yeah no need to worry about this. I didn't look up the other function, it was just from memory. Hard to refactor into common method so forget it.